systemtest/cmd/sendotlp/main.go

```diff
@@ -20,70 +20,147 @@ package main
 	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc"
-	"go.opentelemetry.io/otel/exporters/stdout/stdoutmetric"
-	"go.opentelemetry.io/otel/exporters/stdout/stdouttrace"
 	"go.opentelemetry.io/otel/metric"
 	"go.opentelemetry.io/otel/metric/sdkapi"
 	"go.opentelemetry.io/otel/sdk/export/metric/aggregation"
+	"go.opentelemetry.io/otel/sdk/instrumentation"
 	"go.opentelemetry.io/otel/sdk/metric/aggregator/histogram"
 	controller "go.opentelemetry.io/otel/sdk/metric/controller/basic"
 	"go.opentelemetry.io/otel/sdk/metric/export"
 	processor "go.opentelemetry.io/otel/sdk/metric/processor/basic"
 	"go.opentelemetry.io/otel/sdk/metric/selector/simple"
 	"go.opentelemetry.io/otel/sdk/resource"
 	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+	"go.opentelemetry.io/otel/sdk/trace/tracetest"
 	"go.opentelemetry.io/otel/trace"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+	"go.uber.org/zap/zapgrpc"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/grpclog"
 )
 
+var (
+	endpoint = flag.String(
+		"endpoint",
+		getenvDefault("OTEL_EXPORTER_OTLP_ENDPOINT", "http://localhost:8200"),
+		"target URL to which sendotlp will send spans and metrics ($OTEL_EXPORTER_OTLP_ENDPOINT)",
+	)
+
+	secretToken = flag.String(
+		"secret-token",
+		"",
+		"Elastic APM secret token. Note: setting this overrides $OTEL_EXPORTER_OTLP_HEADERS",
+	)
+
+	logLevel = zap.LevelFlag(
+		"loglevel", zapcore.InfoLevel,
+		"set log level to one of DEBUG, INFO (default), WARN, ERROR, DPANIC, PANIC, FATAL",
+	)
+)
+
+func getenvDefault(key, defaultVal string) string {
+	val := os.Getenv(key)
+	if val != "" {
+		return val
+	}
+	return defaultVal
+}
+
 func main() {
-	if err := Main(context.Background()); err != nil {
-		log.Fatal(err)
+	flag.Parse()
+	zapcfg := zap.NewProductionConfig()
+	zapcfg.EncoderConfig.EncodeTime = zapcore.RFC3339TimeEncoder
+	zapcfg.EncoderConfig.EncodeLevel = zapcore.CapitalColorLevelEncoder
+	zapcfg.Encoding = "console"
+	zapcfg.Level = zap.NewAtomicLevelAt(*logLevel)
+	logger, err := zapcfg.Build()
+	if err != nil {
+		panic(err)
+	}
+	defer logger.Sync()
+	grpclog.SetLogger(zapgrpc.NewLogger(logger, zapgrpc.WithDebug()))
+
+	if err := Main(context.Background(), logger.Sugar()); err != nil {
+		logger.Fatal("error sending data", zap.Error(err))
 	}
 }
 
-func Main(ctx context.Context) error {
-	const otelEndpointEnv = "OTEL_EXPORTER_OTLP_ENDPOINT"
-	endpoint := os.Getenv(otelEndpointEnv)
-	if endpoint == "" {
-		endpoint = "http://localhost:8200"
-		os.Setenv(otelEndpointEnv, endpoint)
+func Main(ctx context.Context, logger *zap.SugaredLogger) error {
+	var insecure bool
+	endpointURL, err := url.Parse(*endpoint)
+	if err != nil {
+		return fmt.Errorf("failed to parse endpoint: %w", err)
+	}
+	switch endpointURL.Scheme {
+	case "http":
+		if endpointURL.Port() == "" {
+			endpointURL.Host = net.JoinHostPort(endpointURL.Host, "80")
+		}
+		insecure = true
+	case "https":
```

Comment:
Since we're parsing the scheme, maybe it also makes sense to allow `https+insecure`? I recently saw it added to the `pprof` command (https://go-review.googlesource.com/c/go/+/33157/2/src/cmd/pprof/internal/fetch/fetch.go) and thought it was handy.

Reply:
I'm trying to keep it as close as possible to the config options supported by OTel SDKs (https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/protocol/exporter.md#configuration-options). `https+insecure` won't be recognised by the SDKs.

I'm not sure how users are expected to disable TLS verification though - might come back to this.

Reply:
I see, thank you for the link to the spec. I hadn't read it